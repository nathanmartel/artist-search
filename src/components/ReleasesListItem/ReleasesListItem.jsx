import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './ReleasesListItem.css';

const ReleasesListItem = ({ id, title }) => {
  
  const albumArt = () => {
    fetch(`http://coverartarchive.org/release/${id}/front`)
      .then(res => { 
        if(res.ok) return `http://coverartarchive.org/release/${id}/front`;
        else return 'https://static1.squarespace.com/static/5d2e2c5ef24531000113c2a4/5d392a924397f100011fa30e/5e3a25cd835c71275a63090f/1584294919598/?format=1500w'
      });
  };
  
  return (
    <li key={id}>
      <Link to={`./${title}`}>
        {/* { console.log('albumArt is', albumArt()) } */}
        <img src={`http://coverartarchive.org/release/${id}/front`} alt={`${title} cover`} />
        <h4>{ title }</h4>
      </Link>
    </li>
  );
};

ReleasesListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default ReleasesListItem;
