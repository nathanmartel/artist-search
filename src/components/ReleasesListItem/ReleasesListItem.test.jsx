import React from 'react';
import { shallow } from 'enzyme';
import ReleasesListItem from './ReleasesListItem';

describe('ReleasesListItem component', () => {
  it('renders ReleasesListItem', () => {
    const wrapper = shallow(<ReleasesListItem id={'idstring'} title={'OK Computer'} coverArt={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
