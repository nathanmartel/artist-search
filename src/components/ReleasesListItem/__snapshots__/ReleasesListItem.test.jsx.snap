// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleasesListItem component renders ReleasesListItem 1`] = `
<li
  key="idstring"
>
  <Link
    to="./OK Computer"
  >
    <img
      alt="OK Computer cover"
      src="http://coverartarchive.org/release/idstring/front"
    />
    <h4>
      OK Computer
    </h4>
  </Link>
</li>
`;
